### PR TITLE
Fix org admin can not access org project unit

### LINKED ISSUE #25670: Org admin users can not access org project
### Description

Thanks @kerwin612 for reporting this issue.

In some special cases, org admin users can not access `org level` project unit.
I also noticed this issue several month ago, but I can't reproduce this issue again, so I didn't report this bug.

But @kerwin612 also reported this issue today, so it seems that this is a reproducible issue.

This issue can be fixed by some magic operations:
- convert the permission into `General Access`, you will see the permission of project unit is `No Access`.
- convert it into `Write`, and save.
- convert the permission into `Administrator Access`, and save.
- Now the team will become a real admin team :)
- Then no matter what permission you set in `General Access`, `Administrator Access` would work well.

~ps: I also can't reproduce this issue now in a new 1.19.3 Gitea instance or an old 1.18.5 instance update to 1.19.3~

Edit:
I misunderstand this issue, admin users can not access org level project unit, not repo level project unit.
I can reproduce this issue now, let's fix it. 😄 

### Gitea Version

1.19.3

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

self host

### Database

None

### PR DESCRIPTION
**Pull request by yp05327**
Fix #25670

We should backport the logic fix to 1.19 and 1.20
DB migration is only for 1.21

### CODE REVIEW

File: models/migrations/v1_21/v268.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		TypeIssues                          // 2 issues
+		TypePullRequests                    // 3 PRs
+		TypeReleases                        // 4 Releases
+		TypeWiki                            // 5 Wiki
+		TypeExternalWiki                    // 6 ExternalWiki
+		TypeExternalTracker                 // 7 ExternalTracker
+		TypeProjects                        // 8 Kanban board
+		TypePackages                        // 9 Packages
+		TypeActions                         // 10 Actions
+	)
```
Comment (puni9869) [inline models/migrations/v1_21/v268.go]: 💯

Comment (yp05327) [reply models/migrations/v1_21/v268.go]: ???

**Comment by yp05327** [comment]
Need someone to review.

File: models/migrations/v1_22/v283.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_21 //nolint
```
Comment (lunny) [inline models/migrations/v1_22/v283.go]:
```suggestion
package v1_22 //nolint
```

**Comment by yp05327** [comment]
Need reviewers.

File: models/migrations/v1_22/v285.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	// find all admin teams
+	teams := make([]*Team, 0)
+	err := sess.Where("team.authorize = ?", AccessModeAdmin).Find(&teams)
```
Comment (lunny) [inline models/migrations/v1_22/v285.go]: It may consume much memory for a big instance.

Comment (yp05327) [reply models/migrations/v1_22/v285.go]: Done.

File: models/migrations/v1_22/v285.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+				}
+				// external unit should be read
+				if u == TypeExternalWiki || u == TypeExternalTracker {
+					newTeamUnit.AccessMode = AccessModeRead
+				} else {
+					newTeamUnit.AccessMode = AccessModeAdmin
+				}
+
+				if existingUnitTypes.Contains(u) {
+					sess.Cols("access_mode").Update(newTeamUnit)
```
Comment (lunny) [inline models/migrations/v1_22/v285.go:119]: returned error haven't been handlered.

File: models/migrations/v1_22/v285.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+				} else {
+					sess.Insert(newTeamUnit)
```
Comment (lunny) [inline models/migrations/v1_22/v285.go:121]: The same as above.

File: models/migrations/v1_22/v285.go [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+			if err := sess.Commit(); err != nil {
+				return err
+			}
+			if err := sess.Begin(); err != nil {
+				return err
+			}
+		}
+	}
+
+	return sess.Commit()
```
Comment (lunny) [inline models/migrations/v1_22/v285.go:136]: You need to detect whether there are still records haven't been committed. Some database will fail if submit a zero change commit.

**Comment by yp05327** [comment]
@lunny 
You added a doctor to fix DB, so I think the migration is not needed.
Just merge the logic fix is OK I think.

**Pull request closed without merging**